ddtrace/debugging/_probe/remoteconfig.py

```diff
@@ -119,13 +121,32 @@ def probe(_id, _type, attribs):
     raise ValueError("Unknown probe type: %s" % _type)
 
 
+_SNAPSHOT_PREFIX = "snapshotProbe_"
+_METRIC_PREFIX = "metricProbe_"
+_CONFIG_REGEX = re.compile(r"^[\da-f]{8}-([\da-f]{4}-){3}[\da-f]{12}$", re.IGNORECASE)
+
+
+def _makeProbes(probes, _type):
+    return [probe(p["id"], _type, p) for p in probes]
+
+
 @_filter_by_env_and_version
-def get_probes(service_config):
-    # type: (dict) -> Iterable[Probe]
-    return chain(
-        [probe(p["id"], "snapshotProbes", p) for p in service_config.get("snapshotProbes") or []],
-        [probe(p["id"], "metricProbes", p) for p in service_config.get("metricProbes") or []],
-    )
+def get_probes(config_id, config):
+    # type: (str, dict) -> Iterable[Probe]
+
+    if config_id.startswith(_SNAPSHOT_PREFIX):
+        return _makeProbes([config], "snapshotProbes")
+
+    if config_id.startswith(_METRIC_PREFIX):
+        return chain(_makeProbes([config], "metricProbes"))
+
+    if _CONFIG_REGEX.match(config_id):
```

Comment:
How big of a deal is if we don't check for the expected pattern here? I'd like to avoid the call to `match` if we can.

Reply:
we can avoid it once we no longer need to support this configuration type. but until then we need to verify we only treat configurations id that are guid as service config.